### ISSUE TITLE
Default value are not detected if they are defined in the parent

### ISSUE THREAD

**Issue by snicoll** [spring-projects/spring-boot#5434]
Consider the following nested class

```
static class Foo {

  private int counter = 42;
  // getter & setter
}
```

and the following extension

```
static class Bar extends Foo {
  private int name;
  // getter setter
```

If a `Bar` instance is used as a target for meta-data generation, the `counter` property is exposed but not the default value. It looks like it only works when `Foo` is exposed.

**Comment by hrishi2323** [comment]
The members from the parent type were not fetched while trying to fetch the fieldValues, hence the default values were not populated for those properties. Moved the method to the TypeElementMembers class and called the fieldValueParser recursively to compute the field values. Submitted a pull request for the fix (#7491).

**Comment by snicoll** [comment]
Closing in favour of the PR ##7491
